cmd/templ/generatecmd/run/run.go

```diff
@@ -2,39 +2,72 @@ package run
 
 import (
 	"context"
-	"fmt"
 	"os"
 	"os/exec"
+	"runtime"
+	"strconv"
 	"strings"
 	"sync"
+	"syscall"
 )
 
 var m = &sync.Mutex{}
 var running = map[string]*exec.Cmd{}
 
+func KillAll() (err error) {
+	m.Lock()
+	defer m.Unlock()
+	for _, cmd := range running {
+		if runtime.GOOS == "windows" {
+			kill := exec.Command("TASKKILL", "/T", "/F", "/PID", strconv.Itoa(cmd.Process.Pid))
+			kill.Stderr = os.Stderr
+			kill.Stdout = os.Stdout
+			err := kill.Run()
+			if err != nil {
+				return err
+			}
+			continue
+		}
+		err := syscall.Kill(-cmd.Process.Pid, syscall.SIGKILL)
+		if err != nil {
+			return err
+		}
+	}
+	running = map[string]*exec.Cmd{}
+	return
+}
+
 func Run(ctx context.Context, workingDir, input string) (cmd *exec.Cmd, err error) {
 	m.Lock()
 	defer m.Unlock()
 	cmd, ok := running[input]
 	if ok {
-		if err = cmd.Process.Kill(); err != nil {
-			return nil, fmt.Errorf("failed to kill existing process: %w", err)
-		}
+		var wg sync.WaitGroup
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			err = KillAll()
+			if err != nil {
+				return
+			}
```

Comment:
Here, we're checking for a return error, but then returning from the go func.

And also, there's only one thing in the waitgroup.

I don't see the point of the go routine and waitgroup, since the code would operate the same without them.

Reply:
I have removed the waitgroup in the latest commit, it works with no problems.